### PR TITLE
[Enhancement] Write >2GB large chunk

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/15316

## Problem Summary(Required) ：

derived from https://github.com/StarRocks/starrocks/pull/19444

when ctrl->attachment is empty, we can write the http rpc like std rpc using proto, refer to `be/src/exec/tablet_sink.cpp`

## Results
set `rpc_http_min_size = 0` to force rpc via http, the msg of inserting data is shown as follow:
```
mysql> insert into test.test_array (select i_manufact_id, array_agg(i_item_sk), array_agg(i_brand_id) from item group by i_manufact_id limit 1000);
Query OK, 995 rows affected (0.20 sec)
{'label':'insert_5c86d254-c238-11ed-9641-0242ca170dfb', 'status':'VISIBLE', 'txnId':'15032'}
```
output msg:

```
W0314 15:18:03.396871 196843 tablet_sink.cpp:250] NodeChannel::_open issue a http rpc, request size = 361
W0314 15:18:03.510782 197267 sink_buffer.cpp:402] issue a http rpc, attachment's size = 334873 , total size = 334991
W0314 15:18:03.524915 197188 tablet_sink.cpp:595] NodeChannel::_send_request issue a http rpc, request size = 557717
```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature [manully]
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1899]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1899]|

File: be/src/exec/pipeline/exchange/sink_buffer.cpp [hunk 1]
```diff
@@ -370,20 +371,47 @@ Status SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id, const std::fun
...
+        request.params->SerializeToZeroCopyStream(&wrapper);
+        // append params to iobuf
+        size_t params_size = iobuf.size();
+        closure->cntl.request_attachment().append(&params_size, sizeof(params_size));
+        closure->cntl.request_attachment().append(iobuf);
+        // append attachment
+        size_t attachment_size = request.attachment.size();
+        closure->cntl.request_attachment().append(&attachment_size, sizeof(attachment_size));
+        closure->cntl.request_attachment().append(request.attachment);
+        LOG(WARNING) << "issue a http rpc, attachment's size = " << attachment_size
```
Comment (imay) [inline be/src/exec/pipeline/exchange/sink_buffer.cpp:404]: I think this log is not necessary should be deleted

File: be/src/exec/tablet_sink.cpp [hunk 2]
```diff
@@ -553,16 +563,41 @@ Status NodeChannel::_send_request(bool eos, bool wait_all_sender_close) {
...
         DCHECK(request.requests_size() == 1);
-        _stub->tablet_writer_add_chunk(&_add_batch_closures[_current_request_index]->cntl, request.mutable_requests(0),
-                                       &_add_batch_closures[_current_request_index]->result,
-                                       _add_batch_closures[_current_request_index]);
+        if (UNLIKELY(request.ByteSizeLong() > _parent->_rpc_http_min_size)) {
+            TNetworkAddress brpc_addr;
+            brpc_addr.hostname = _node_info->host;
+            brpc_addr.port = _node_info->brpc_port;
+            _add_batch_closures[_current_request_index]->cntl.http_request().set_content_type("application/proto");
+            auto http_stub = BrpcStubCache::create_http_stub(brpc_addr);
```
Comment (chaoyli) [inline be/src/exec/tablet_sink.cpp:591]: need to check whether the http_stub is a nullptr or not.

File: be/src/exec/pipeline/exchange/sink_buffer.cpp (hunk 1 above)
Comment (fzhedu) [reply be/src/exec/pipeline/exchange/sink_buffer.cpp]: change to vlog

File: be/src/exec/pipeline/exchange/exchange_sink_operator.cpp [hunk 3]
```diff
@@ -278,8 +278,8 @@ Status ExchangeSinkOperator::Channel::send_chunk_request(RuntimeState* state, PT
         delta_statistic->to_pb(chunk_request->mutable_query_statistics());
     }
 
-    TransmitChunkInfo info = {this->_fragment_instance_id, _brpc_stub, std::move(chunk_request), attachment,
-                              attachment_physical_bytes};
+    TransmitChunkInfo info = {this->_fragment_instance_id, _brpc_stub,     std::move(chunk_request), attachment,
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/exec/pipeline/exchange/exchange_sink_operator.cpp]: formated

File: be/src/exec/tablet_sink.cpp (hunk 2 above)
Comment (fzhedu) [reply be/src/exec/tablet_sink.cpp]: updated.

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by fzhedu** [comment]
@mergify backport-2.5

**Comment by fzhedu** [comment]
@mergify backport branch-2.5

**Comment by fzhedu** [comment]
@mergify backport branch-3.0

**Pull request merged**